tests/__init__.py

```diff
@@ -406,10 +513,97 @@ def test_dotd(self):
                 ("--version",),
             )
 
+    def test_dotd_ge_16(self):
+        """Test environment with common ``*.d`` directories.
+        Only for versions greater or equal to 16"""
+        for sub_env in matrix(odoo_skip={"11.0", "12.0", "13.0", "14.0", "15.0"}):
```

Comment:
Please avoid duplicating all these stuff. You can do inside the same test several loops, letting the common ones together, and just including the differences in different loops.

Reply:
Adding new commits amending code from this same PR makes the review harder. Better to cleanup the commits that add the original code.